data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/BeaconRestApiTest.java

```diff
@@ -57,4 +58,12 @@ public void RestApiShouldHaveVersionEndpoint() throws Exception {
 
     verify(mockApp).get(eq(VersionHandler.ROUTE), any(VersionHandler.class));
   }
+
+  @Test
+  public void restApiShouldHaveBeaconHeadEndpoint() throws Exception {
```

Comment:
I'm not sure these tests are providing any real value.  They aren't checking it's actually wired up correctly and are essentially just duplicating the internal logic of the class being tested.

If you can get a list of registered routes from the server and compare to the expected set that could be useful and lower maintenance but the risk of not registering the API at all seems fairly low.

Reply:
It's checking that the route has been linked to the correct class during startup. I added these because I'm working on the assumption that this is my altitude for testing - the object is handling correctly, and its linked to javalin, and beyond that javalin does its job...

Reply:
if this was not wired in right, and this test fails, would ALL the tests fail?

Reply:
depends on the test. this specific case is only looking at where this specific route is wired, so just this would fail... If we break the whole app object, then yes a lot of tests might fail